python/cairo-ec/tests/circuits/test_circuits.py

```diff
@@ -469,3 +474,224 @@ def test_assert_is_on_curve(
+                "div_d_coeff_1": int(rlcSumDlogDiv.b.denominator[1].value),
+                "div_d_coeff_2": int(rlcSumDlogDiv.b.denominator[2].value),
+                "div_d_coeff_3": int(rlcSumDlogDiv.b.denominator[3].value),
+                "div_d_coeff_4": int(rlcSumDlogDiv.b.denominator[4].value),
+                "div_d_coeff_5": int(rlcSumDlogDiv.b.denominator[5].value),
+                "div_d_coeff_6": int(rlcSumDlogDiv.b.denominator[6].value),
+                "div_d_coeff_7": int(rlcSumDlogDiv.b.denominator[7].value),
+                "div_d_coeff_8": int(rlcSumDlogDiv.b.denominator[8].value),
+                "g_x": int(points[0].x),
+                "g_y": int(points[0].y),
+                "r_x": int(points[1].x),
+                "r_y": int(points[1].y),
+                "ep1_low": int(epns_low[0][0]),
+                "en1_low": int(epns_low[0][1]),
+                "sp1_low": int(epns_low[0][2] % curve.p),
+                "sn1_low": int(epns_low[0][3] % curve.p),
+                "ep2_low": int(epns_low[1][0]),
+                "en2_low": int(epns_low[1][1]),
+                "sp2_low": int(epns_low[1][2] % curve.p),
+                "sn2_low": int(epns_low[1][3] % curve.p),
+                "ep1_high": int(epns_high[0][0]),
+                "en1_high": int(epns_high[0][1]),
+                "sp1_high": int(epns_high[0][2] % curve.p),
+                "sn1_high": int(epns_high[0][3] % curve.p),
+                "ep2_high": int(epns_high[1][0]),
+                "en2_high": int(epns_high[1][1]),
+                "sp2_high": int(epns_high[1][2] % curve.p),
+                "sn2_high": int(epns_high[1][3] % curve.p),
+                "q_low_x": int(q_low.x),
+                "q_low_y": int(q_low.y),
+                "q_high_x": int(q_high.x),
+                "q_high_y": int(q_high.y),
+                "q_high_shifted_x": int(q_high_shifted.x),
+                "q_high_shifted_y": int(q_high_shifted.y),
+                "a0_x": int(a0.x),
+                "a0_y": int(a0.y),
+                "a": int(curve.a),
+                "b": int(curve.b),
+                "base_rlc": int(rlc_coeff),
+            }
+
+            if prime == curve.p:
+                cairo_run("ecip_2p", **inputs)
+                compiled_circuit = circuit_compile(cairo_program, "ecip_2p")
+                values_ptr = flatten(compiled_circuit["constants"]) + [
+                    limb for v in inputs.values() for limb in int_to_uint384(v)
+                ]
+                cairo_run(
+                    "test__circuit",
+                    values_ptr=values_ptr,
+                    values_ptr_len=len(values_ptr),
+                    p=int_to_uint384(curve.p),
+                    **compiled_circuit,
+                )
+                cairo_run(
+                    "ecip_2p_compiled",
+                    **{k: int_to_uint384(v) for k, v in inputs.items()},
+                    p=int_to_uint384(curve.p),
+                )
```

Comment:
this never runs in the CI

```suggestion
            if prime == curve.p:
                cairo_run("ecip_2p", **inputs)
                compiled_circuit = circuit_compile(cairo_program, "ecip_2p")
                values_ptr = flatten(compiled_circuit["constants"]) + [
                    limb for v in inputs.values() for limb in int_to_uint384(v)
                ]
              cairo_run(
                  "test__circuit",
                  values_ptr=values_ptr,
                  values_ptr_len=len(values_ptr),
                  p=int_to_uint384(curve.p),
                  **compiled_circuit,
              )
              cairo_run(
                  "ecip_2p_compiled",
                  **{k: int_to_uint384(v) for k, v in inputs.items()},
                  p=int_to_uint384(curve.p),
              )
```